engine/crates/composition/src/subgraphs/fields.rs

```diff
@@ -100,23 +100,14 @@ impl<'a> FieldWalker<'a> {
         self.field().arguments.iter().map(move |id| self.walk(*id))
     }
 
-    /// Returns true iff there is an `@key` directive containing exactly this field (no composite
-    /// key).
-    pub fn is_key(self) -> bool {
+    /// Returns true iff there is an `@key` directive containing this field, possibly with others
+    /// as part of a composite key.
+    pub(crate) fn is_part_of_key(self) -> bool {
```

Comment:
Note that there's nothing about nesting in the implementation. What does it mean for a nested field to be part of the key (in another value type or entity)? For example:

```graphql
# First subgraph

type A @key("b { id } c") {
  b: B!
  c: String!
}

type B {
  id: ID!
  foo: String
}

# Other subgraph

type B @key("id") {
  id: ID!
  bar: String
}
```

Is that valid?

Reply:
Apparently it should be.

Reply:
Implemented.